core/src/main/java/org/infinispan/marshall/protostream/impl/SerializationContextRegistryImpl.java

```diff
@@ -92,11 +109,20 @@ private void update(MarshallerType type, Consumer<MarshallerContext> consumer) {
       }
    }
 
+   private static void register(SerializationContextInitializer sci, SerializationContext... ctxs) {
+      for (SerializationContext ctx : ctxs) {
+         sci.registerSchema(ctx);
+         sci.registerMarshallers(ctx);
+      }
+   }
+
+   // Required until IPROTO-136 is resolved to ensure that custom marshaller implementations are not overridden by
+   // non-core modules registering their SerializationContextInitializer(s) which depend on a core initializer.
```

Comment:
@anistor I remembered why `MarshallerContext` was required. Without it we get errors in non-core modules as the marshaller overriding is lost, hence we have to ensure in the registry that the marshallers are always registered last.

https://ci.infinispan.org/job/Infinispan/view/change-requests/job/PR-8187/1/